src/components/letter/Letter.tsx

```diff
@@ -0,0 +1,101 @@
+import { graphql, PageProps } from "gatsby";
+import React from "react";
+import { LetterData } from "../../types/EmailData";
+import { LetterProps, LetterConfig } from "../../types/PropTypes";
+import Layout from "../common/Layout";
+import EmailList from "../email-list/EmailList";
+import LetterForm from "./LetterForm";
+import { OfficialRestrict } from "./types";
+
+/**
+ * A rendered email, containing links to send or copy.
+ */
+export default class Letter extends React.Component<PageProps<LetterProps>> {
+  letterData: LetterData;
+  title: string;
+  meta: string;
+  siteConfig: LetterConfig;
+
+  /**
+   * Initialize the component and its state.
+   * @param {PageProps<EmailProps>} props
+   */
+  constructor(props: PageProps<LetterProps>) {
+    super(props);
+    this.siteConfig = this.props.data.siteConfig;
+    this.letterData = this.props.data.markdownRemark.frontmatter;
+    this.title = `Defund 12 in ${this.letterData.city}, ${this.letterData.state}`;
+    this.meta = `Send a pre-written letter directly to ${this.letterData.city}, ${this.letterData.state} officials`;
+  }
+
+  /**
+   * React render method.
+   * @return {React.ReactNode} the rendered component
+   */
+  render(): React.ReactNode {
+    // parse the offical restricts from the template markdown that look like
+    // locality:legislatorUpperBody into structured restricts for our APIs
+    const officialRestricts = this.letterData.officials?.map(
```

Comment:
That's not the react style I'm used to. I'm used to only putting initializations in componentDidMount that are heavy, and everything else that's setting up the logic you need for render ... goes in render. Also, componentDidMount is going away with functional components (it becomes "useEffect" which is pretty clear it's meant for side-effects, not calculations).

It's only going to be called once because the props that are being passed in aren't changing so it's not a performance issue